launcher/launcher.go

```diff
@@ -239,6 +242,10 @@ func (l *launcher) reconfigureWait(timeout time.Duration) (*config.C, error) {
 				err := l.validator.Validate(update)
 				if err != nil {
 					l.log.Errorf("Config update validation failed: %v", err)
+					heatlhErr := &cb_errors.BeaterUnhealthyError{}
+					if errors.As(err, heatlhErr) {
```

Comment:
Let's test this new functionality, it can be a part of `TestLauncherValidator`

Reply:
Also, note that you are checking the error that returned from the `Validator` but `BeaterUnhealthyError` error is actually returned from `config.New` and wrapped in the validator and probably won't get detected here.

Reply:
> Let's test this new functionality, it can be a part of `TestLauncherValidator`

Sure



> Also, note that you are checking the error that returned from the `Validator` but `BeaterUnhealthyError` error is actually returned from `config.New` and wrapped in the validator and probably won't get detected here.

The [test](https://github.com/elastic/cloudbeat/pull/583/files#diff-5a256ba4448e51ae85f5cdae75f5093675bc7ea190d00bcf687fffa782822ca8R31) for the errors pkg checking exactly this. First create the `BeaterUnhealthyError` and than wrap it, later test that using `errors.As` it will get the original error.

From golang https://pkg.go.dev/errors#As
> As finds the first error in err's chain that matches target, and if one is found, sets target to that error value and returns true. Otherwise, it returns false.